### PR TITLE
feat(Switch): refactor to typescript

### PR DESCRIPTION
**Pull request by AadneRo**

### CODE REVIEW

File: packages/dnb-eufemia/src/components/switch/Switch.tsx [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+      if (innerRef.current) {
+        innerRef.current.focus()
+      }
+    },
+    [callOnChange, innerRef, isChecked, onChangeEnd, readOnly]
+  )
+
+  const onKeyDownHandler = useCallback(
+    (event) => {
+      switch (keycode(event)) {
```
Comment (joakbjerk) [inline packages/dnb-eufemia/src/components/switch/Switch.tsx:125]: Should probably be replaced with `event.key` or `event.code`, so we can get rid of the `keycode` dependency 🤔 and then change the case to `case 'Enter' to make it work

File: packages/dnb-eufemia/src/components/switch/SwitchDocs.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import { PropertiesTableProps } from '../../shared/types'
+
+export const SwitchProperties: PropertiesTableProps = {
+  checked: {
+    doc: 'determine whether the switch is checked or not. The default will be `false`.',
```
Comment (joakbjerk) [inline packages/dnb-eufemia/src/components/switch/SwitchDocs.ts:5]: as Tobias reminded me, the start of the doc texts should be capitalized from now on 😄 

I used AI to quickly add the capitalized changes when I did similar work on the `Accordion` docs

**Review by joakbjerk — APPROVED** [review]
Looks good to me 👍

But we should probably add the noted changes before merge

File: packages/dnb-eufemia/src/components/switch/types.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    | ((elem: HTMLInputElement) => void)
+} & Omit<
+  React.HTMLProps<HTMLElement>,
+  'ref' | 'size' | 'onChange' | 'innerRef'
+> &
+  SpacingProps &
+  DepricatedSwitchProps
+
+// deprecated, can be removed in v11
+type DepricatedSwitchProps = {
```
Comment (langz) [inline packages/dnb-eufemia/src/components/switch/types.ts:95]:
```suggestion
type DeprecatedSwitchProps = {
```

File: packages/dnb-eufemia/src/components/switch/SwitchDocs.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    type: ['default', 'medium', 'large'],
+    status: 'optional',
+  },
+  status: {
+    doc: 'Text with a status message. The style defaults to an error message. You can use `true` to only get the status color, without a message.',
+    type: 'ReactNode',
+    status: 'optional',
+  },
+  statusState: {
+    doc: 'Defines the state of the status. Currently, there are two statuses `[error, info]`. Defaults to `error`.',
```
Comment (langz) [inline packages/dnb-eufemia/src/components/switch/SwitchDocs.ts:40]: Note to self, I think we should update/change this in many places in our docs, as the type says `type: ['error', 'warn', 'info', 'success', 'marketing']`, so it got to be more that only two statuses 🤔 

Nothing to improve in this PR, I'll look into it after this PR is merged.

**Review by langz — APPROVED** [review]

**Comment by joakbjerk** [comment]
I think this one is ready for merge now 🤔

**Review by tujoworker — APPROVED** [review]
Looks good 👍

**Pull request merged**